### ISSUE TITLE
Develop chrome extension with Next.js 

### ISSUE THREAD

**Issue by Nishchit14** [zeit/next.js#2096]
Next.js seems prominent framework for ReactJS. It fulfills most of the headaches like SSR, Routing, Hot reload etc.
Is there any example to create Chrome Extension with Next.js?

**Comment by notrab** [comment]
Are there any benefits to using Next.js vs. create-react-app for an extension? I don't know what advantages SSR would give in an extension. I'm new to creating extensions for Chrome but I'd love to hear more of your thoughts on this.

**Comment by timneutkens** [comment]
I don't think you can run a full webserver in a chrome extension, can you? 🤔

**Comment by timneutkens** [comment]
https://stackoverflow.com/questions/5053244/is-it-possible-to-embed-a-http-server-in-a-google-chrome-extension

**Comment by sergiodxa** [comment]
🤔 if it's not possible because of the http server then as with Electron you can use the statics export of v3 to do it.

**Comment by albinekb** [comment]
Come to the community slack, GitHub issues is not a place for these discussions

**Comment by sergiodxa** [comment]
@albinekb why do you think that? the issues can be checked in the future if someone want to do the same thing, the community slack has a limit of messages and it start hiding old ones :(

**Comment by arunoda** [comment]
With [static export](https://zeit.co/blog/next3-preview#next-export), you could get our other features beside SSR. (like easy routing, etc)

And I like the idea of @sergiodxa. Issue are always better if they are focused issues like this.
To fix a issue in a app (not necessarily, Next.js issue) then the slack would be fine.

**Comment by Nishchit14** [comment]
Thanks, @arunoda, `static-export` will work in this case. 
But one thing is, Can we **watch** the build while static export?

**Comment by sedubois** [comment]
+1 for HMR with static exports :)

**Comment by arunoda** [comment]
@Nishchit14 there won't be any watching support for static export. Only HMR support is our normal HMR when run with the dev mode.

We've only minor bugs to fix. Within 1-2 weeks max.

**Comment by sedubois** [comment]
@arunoda HMR can be helpful with static exports because there are issues which are specific to static exports. E.g you can't detect Accept-Language in the same way as there's no HTTP request. Authentication also needs to be handled differently. So as I'm trying to adapt my dynamic Next.js to use next export, I constantly need to run next build + next export after each code change, which is quite cumbersome.

I understand if other things have more priority though, just mentioning that without HMR it's not optimal and hopefully could be improved in future.

(NB: I think create-React-app + React-snapshot allows for HMR even with static pages.)

**Comment by arunoda** [comment]
@sedubois Let me rephrase about the HMR support in Next.js

**Because some people would realize you won't get HMR support with Next.js even for static pages.**

Here's the how your `package.json` file looks like:

```js
{
  "dev": "next",
  "build": "next build",
  "export": "next export"
}
```

1. Develop your app with running `npm run dev`. Then you'll get HMR support for your app just like any other Next.js app.
2. Add the pages you want to export in `next.config.js`
3. When you want to deploy your app, do this: `npm run build && npm run export`

Then you'll get a static directory where you can deploy anywhere.

> you can't detect Accept-Language in the same way as there's no HTTP request
If you want to do that, you want to go for a normal Next.js app. Or you can wrap your static deployment with a proxy and do stuff there.

> Authentication also needs to be handled differently
Totally. This is an static app. But running that with `npm run dev` doesn't break anything.

> (NB: I think create-React-app + React-snapshot allows for HMR even with static pages.)
No tool is good for evert job. That's why we've so many solutions for a given problem. Use a tool that's works best for you.

**Comment by sedubois** [comment]
Thanks @arunoda for the clarification. Similar observation with server-side redirection which can't be done with static exports.

**Comment by arunoda** [comment]
> Similar observation with server-side redirection which can't be done with static exports.

That's why they called static apps :D

**Comment by sedubois** [comment]
Yes of course 🙂 My point was `next export` with HMR would have been useful to test these kinds of differences.
